src/rgw/rgw_rados.cc

```diff
@@ -7531,29 +7529,45 @@ int RGWRados::raw_obj_stat(rgw_raw_obj& obj, uint64_t *psize, real_time *pmtime,
   map<string, bufferlist> unfiltered_attrset;
   uint64_t size = 0;
   struct timespec mtime_ts;
+  int prefetch_ret = 0;
+  uint64_t prefetch_offset = 0;
 
   ObjectReadOperation op;
   if (objv_tracker) {
     objv_tracker->prepare_op_for_read(&op);
   }
   if (attrs) {
-    op.getxattrs(&unfiltered_attrset, NULL);
+    if (prefetch && first_chunk) {
+      cls_rgw_head_prefetch(op, prefetch->off, prefetch->len,
+                            cct->_conf->rgw_max_chunk_size,
+                            &prefetch_ret, &prefetch_offset, first_chunk, &unfiltered_attrset);
+    } else {
+      op.getxattrs(&unfiltered_attrset, NULL);
+    }
   }
   if (psize || pmtime) {
     op.stat2(&size, &mtime_ts, NULL);
   }
-  if (first_chunk) {
-    op.read(0, cct->_conf->rgw_max_chunk_size, first_chunk, NULL);
+  if (prefetch && first_chunk && !attrs) {
+    cls_rgw_head_prefetch(op, prefetch->off, prefetch->len,
+                          cct->_conf->rgw_max_chunk_size,
+                          &prefetch_ret, &prefetch_offset, first_chunk, nullptr);
   }
   bufferlist outbl;
   r = rgw_rados_operate(ref.pool.ioctx(), ref.obj.oid, &op, &outbl, null_yield);
 
+  if (prefetch)
+    prefetch->off = prefetch_offset;
+
   if (epoch) {
     *epoch = ref.pool.ioctx().get_last_version();
   }
 
   if (r < 0)
     return r;
+  if (prefetch_ret < 0)
+    return prefetch_ret;
+  // TODO: make use of prefetch_offset
```

Comment:
you can remove this comment. we set `prefetch->off = prefetch_offset;` above, and use that updated offset for the comparison in `RGWRados::get_obj_iterate_cb()`

Reply:
Done